compose_local.go

```diff
@@ -0,0 +1,399 @@
+	Format(parts ...string) string
+}
+
+type composeVersion1 struct {
+}
+
+func (c composeVersion1) Format(parts ...string) string {
+	return strings.Join(parts, "_")
+}
+
+type composeVersion2 struct {
+}
+
+func (c composeVersion2) Format(parts ...string) string {
+	return strings.Join(parts, "-")
+}
+
+// LocalDockerCompose represents a Docker Compose execution using local binary
+// docker-compose or docker-compose.exe, depending on the underlying platform
+type LocalDockerCompose struct {
+	ComposeVersion
+	*LocalDockerComposeOptions
+	Executable           string
+	ComposeFilePaths     []string
+	absComposeFilePaths  []string
+	Identifier           string
+	Cmd                  []string
+	Env                  map[string]string
+	Services             map[string]interface{}
+	waitStrategySupplied bool
+	WaitStrategyMap      map[waitService]wait.Strategy
+}
+
+type (
+	// LocalDockerComposeOptions defines options applicable to LocalDockerCompose
+	LocalDockerComposeOptions struct {
+		Logger Logging
+	}
+
+	// LocalDockerComposeOption defines a common interface to modify LocalDockerComposeOptions
+	// These options can be passed to NewLocalDockerCompose in a variadic way to customize the returned LocalDockerCompose instance
+	LocalDockerComposeOption interface {
+		ApplyToLocalCompose(opts *LocalDockerComposeOptions)
+	}
+
+	// LocalDockerComposeOptionsFunc is a shorthand to implement the LocalDockerComposeOption interface
+	LocalDockerComposeOptionsFunc func(opts *LocalDockerComposeOptions)
+)
+
+func (f LocalDockerComposeOptionsFunc) ApplyToLocalCompose(opts *LocalDockerComposeOptions) {
+	f(opts)
+}
+
+// Down executes docker-compose down
+func (dc *LocalDockerCompose) Down() ExecError {
+	return executeCompose(dc, []string{"down", "--remove-orphans", "--volumes"})
+}
+
+func (dc *LocalDockerCompose) getDockerComposeEnvironment() map[string]string {
+	environment := map[string]string{}
+
+	composeFileEnvVariableValue := ""
+	for _, abs := range dc.absComposeFilePaths {
+		composeFileEnvVariableValue += abs + string(os.PathListSeparator)
+	}
+
+	environment[envProjectName] = dc.Identifier
+	environment[envComposeFile] = composeFileEnvVariableValue
+
+	return environment
+}
+
+func (dc *LocalDockerCompose) containerNameFromServiceName(service, separator string) string {
+	return dc.Identifier + separator + service
+}
+
+func (dc *LocalDockerCompose) applyStrategyToRunningContainer() error {
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return err
+	}
+
+	cli.NegotiateAPIVersion(context.Background())
+
+	for k := range dc.WaitStrategyMap {
+		containerName := dc.containerNameFromServiceName(k.service, "_")
+		composeV2ContainerName := dc.containerNameFromServiceName(k.service, "-")
+		f := filters.NewArgs(
+			filters.Arg("name", containerName),
+			filters.Arg("name", composeV2ContainerName),
+			filters.Arg("name", k.service))
+		containerListOptions := types.ContainerListOptions{Filters: f}
```

Comment:
@baez90 we are going to merge https://github.com/testcontainers/testcontainers-go/pull/514, and I think this line would need to be updated.

Can you double check this line: https://github.com/testcontainers/testcontainers-go/pull/514/files#diff-4c4f2f19edfc5e047aef7e584ae2608724405f66563c6ee3b9957a19febada2bR183

Once merged, could you double check if the new test in it should be updated? 🙏

Reply:
Do you mind if I rebase my PR? I'll resolve the conflicts and make sure the tests are working again and force push afterwards without new changes?